src/Impostor.Server/Recorder/PacketRecorder.cs

```diff
@@ -35,17 +39,32 @@ public PacketRecorder(ILogger<PacketRecorder> logger, IOptions<DebugConfig> opti
 
             _channel = Channel.CreateUnbounded<byte[]>(new UnboundedChannelOptions
             {
-                SingleReader = true,
-                SingleWriter = false,
+                SingleReader = true, SingleWriter = false,
             });
         }
 
         protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
+            _time = DateTimeOffset.UtcNow;
             _logger.LogInformation("PacketRecorder is enabled, writing packets to {0}.", _path);
 
             var writer = File.Open(_path, FileMode.CreateNew, FileAccess.Write, FileShare.Read);
 
+            var context = _pool.Get();
+
+            try
+            {
+                context.Writer.Write(ProtocolVersion);
```

Comment:
Please put in a method `WriteHeader`.

Reply:
There is already a method named WriteHeader, so WriteFileHeader/WriteSessionHeader?